src/node_os.cc

```diff
@@ -151,6 +151,14 @@ static void GetTotalMemory(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+static void GetConstrainedMemory(const FunctionCallbackInfo<Value>& args) {
+  double amount = uv_get_constrained_memory();
+  if (amount < 0)
```

Comment:
This condition is never true because `uv_get_constrained_memory()` returns an unsigned integer. Assigning it to a `double` isn't really correct either because its value may be well outside what can fit in 53 bits.

Reply:
Ah that makes sense. That being said, do we need to also update `GetFreeMemory` and `GetTotalMemory`? This is implemented almost identically to those.

```
static void GetFreeMemory(const FunctionCallbackInfo<Value>& args) {
  double amount = uv_get_free_memory();
  if (amount < 0)
    return;
  args.GetReturnValue().Set(amount);
}
```

Reply:
Ideally yes, but it's less acute for those functions because systems generally don't have > 2**53 bytes of memory.

However, it's common with cgroups to set the memory limit to 2**63 to indicate it's unconstrained and that's what `uv_get_constrained_memory()` reports.

Reply:
ah yea that makes sense. Any recommendations on how best to handle this? Should we expose as a BigInt here?